modules/REST/API.php

```diff
@@ -48,5 +55,8 @@ public function __construct( PLL_Model $model ) {
 	public function init(): void {
 		$this->languages = new V1\Languages( $this->model->languages, $this->model->translatable_objects );
 		$this->languages->register_routes();
+
+		$this->options = new V1\Options( $this->model->options );
+		$this->options->register_routes();
```

Comment:
Can `register_route()` return `\Options` object? And this line can be removed.
Same for languages.

Reply:
`WP_REST_Controller::register_routes()` doesn't return anything. On the other hand, the doc doesn't provide any `@return` tag, so I guess it can return anything?

Reply:
However, now we have this:

```
 ------ ---------------------------------------------------------------------------------- 
  Line   modules/REST/API.php                                                              
 ------ ---------------------------------------------------------------------------------- 
  :28    Property WP_Syntex\Polylang\REST\API::$languages is never read, only written.     
         💡 See: https://phpstan.org/developing-extensions/always-read-written-properties   
  :35    Property WP_Syntex\Polylang\REST\API::$options is never read, only written.       
         💡 See: https://phpstan.org/developing-extensions/always-read-written-properties   
 ------ ----------------------------------------------------------------------------------
```

But I notices something: in the class `API` the properties are `public $model;`, `private $languages`, and `private $options`. I think they should be the opposite:
- `private $model;` because we don't need to access it from the outside,
- `public $languages;` and `public $options;`, so their hooks can be manipulated from the outside.

And by doing so, PHPStan doesn't trigger the previous errors.
What do you think?

Reply:
I forgot that `register_routes()` is defined in `WP_REST_Controller`... 
But your suggestion sounds good!

Reply:
Done in [509c086](https://github.com/polylang/polylang/pull/1569/commits/509c0862442619a8f7d89b472d4cb95ed2295dff) and [1ee0c2b](https://github.com/polylang/polylang/pull/1569/commits/1ee0c2bda35b1f054b1338a8c936f370d36285bd).